src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx

```diff
@@ -171,11 +171,18 @@ export class DashboardAppController {
       dashboardStateManager.getIsViewMode() &&
       !dashboardConfig.getHideWriteControls();
 
-    const getIsEmptyInReadonlyMode = () =>
-      !dashboardStateManager.getPanels().length &&
-      !getShouldShowEditHelp() &&
-      !getShouldShowViewHelp() &&
-      dashboardConfig.getHideWriteControls();
+    const getIsEmptyInReadonlyModeOrUserHasNoPermissions = () => {
```

Comment:
This method name is getting long enough that it's difficult to read...

Would changing it to something like `shouldShowErrorMessage()` be an accurate alternative?

Reply:
Hm, it's not an error state though, it's a valid state, somewhat edge-casey, but still valid.

Reply:
Maybe `shouldShowUnauthorizedEmptyState`?

Also, not really critical to make this change if you like longer function names 🤷‍♂